packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// Default Google Play Store implementation states that reflect the current
+/// installation status of a deferred component module.
+///
+/// Returned by [DeferredComponent.getDeferredComponentInstallState].
```

Comment:
That method seems to return a String instead of one of these enum values?

Reply:
(see answer about custom impls below)

This API can totally still be locked down further if that is determined to be best.

Reply:
If we keep it that way, we should update this doc to no longer say that these enum values are returned by getDeferredComponentInstallState. Nevertheless, I find it strange that we have this enum, but it isn't used anywhere. What is a developer supposed to do with this enum?